pkg/rulebindingmanager/types/v1/types.go

```diff
@@ -52,3 +53,62 @@ func (r *RuntimeAlertRuleBindingRule) Equal(other *RuntimeAlertRuleBindingRule)
 	}
 	return true
 }
+
+var _ runtime.Object = &RuntimeAlertRuleBinding{}
+
+func (r *RuntimeAlertRuleBinding) DeepCopyObject() runtime.Object {
+	if c := r.DeepCopy(); c != nil {
+		return c
+	}
+	return nil
+}
+
+func (r *RuntimeAlertRuleBinding) DeepCopy() *RuntimeAlertRuleBinding {
+	if r == nil {
+		return nil
+	}
+	out := new(RuntimeAlertRuleBinding)
+	r.DeepCopyInto(out)
+	return out
+}
+
+func (r *RuntimeAlertRuleBinding) DeepCopyInto(out *RuntimeAlertRuleBinding) {
+	*out = *r
+	out.TypeMeta = r.TypeMeta
+	out.ObjectMeta = *r.ObjectMeta.DeepCopy()
+	out.Spec = r.Spec.DeepCopy()
+}
+
+func (s *RuntimeAlertRuleBindingSpec) DeepCopy() RuntimeAlertRuleBindingSpec {
+	if s == nil {
+		return RuntimeAlertRuleBindingSpec{}
+	}
+	out := *s
+	if s.Rules != nil {
+		out.Rules = make([]RuntimeAlertRuleBindingRule, len(s.Rules))
+		for i := range s.Rules {
+			out.Rules[i] = s.Rules[i].DeepCopy()
+		}
+	}
+	out.PodSelector = *s.PodSelector.DeepCopy()
+	out.NamespaceSelector = *s.NamespaceSelector.DeepCopy()
+	return out
+}
+
+func (r *RuntimeAlertRuleBindingRule) DeepCopy() RuntimeAlertRuleBindingRule {
+	if r == nil {
+		return RuntimeAlertRuleBindingRule{}
+	}
+	out := *r
+	if r.Parameters != nil {
+		out.Parameters = make(map[string]interface{}, len(r.Parameters))
+		for key, val := range r.Parameters {
+			out.Parameters[key] = val
```

Comment:
Are we okay with this shallow copy? we don't need deep copy as well?

Reply:
I'm removing these changes, I don't need them since we're forced to use dynamic client for the RuntimeAlertRuleBinding